doc/api/fs.md

```diff
@@ -13,1160 +13,1284 @@ way modeled on standard POSIX functions.
+  `-Infinity`, an `Error` will be thrown.
 
-### Event: `'close'`
+### `fsPromises.writeFile(file, data[, options])`
 <!-- YAML
-added: v0.1.93
+added: v10.0.0
+changes:
+  - version: v15.2.0
+    pr-url: https://github.com/nodejs/node/pull/35993
+    description: The options argument may include an AbortSignal to abort an
+                 ongoing writeFile request.
+  - version: v14.12.0
+    pr-url: https://github.com/nodejs/node/pull/34993
+    description: The `data` parameter will stringify an object with an
+                 explicit `toString` function.
+  - version: v14.0.0
+    pr-url: https://github.com/nodejs/node/pull/31030
+    description: The `data` parameter won't coerce unsupported input to
+                 strings anymore.
 -->
 
-Emitted when the `WriteStream`'s underlying file descriptor has been closed.
+* `file` {string|Buffer|URL|FileHandle} filename or `FileHandle`
+* `data` {string|Buffer|Uint8Array|Object}
+* `options` {Object|string}
+  * `encoding` {string|null} **Default:** `'utf8'`
+  * `mode` {integer} **Default:** `0o666`
+  * `flag` {string} See [support of file system `flags`][]. **Default:** `'w'`.
+  * `signal` {AbortSignal} allows aborting an in-progress writeFile
+* Returns: {Promise}
 
-### Event: `'open'`
-<!-- YAML
-added: v0.1.93
--->
+Asynchronously writes data to a file, replacing the file if it already exists.
+`data` can be a string, a {Buffer}, or an object with an own `toString` function
+property. The promise is resolved with no arguments upon success.
 
-* `fd` {integer} Integer file descriptor used by the `WriteStream`.
+The `encoding` option is ignored if `data` is a buffer.
 
-Emitted when the `WriteStream`'s file is opened.
+If `options` is a string, then it specifies the encoding.
 
-### Event: `'ready'`
-<!-- YAML
-added: v9.11.0
--->
+Any specified {FileHandle} has to support writing.
 
-Emitted when the `fs.WriteStream` is ready to be used.
+It is unsafe to use `fsPromises.writeFile()` multiple times on the same file
+without waiting for the promise to be settled.
 
-Fires immediately after `'open'`.
+Similarly to `fsPromises.readFile` - `fsPromises.writeFile` is a convenience
+method that performs multiple `write` calls internally to write the buffer
+passed to it. For performance sensitive code consider using
+[`fs.createWriteStream()`][].
 
-### `writeStream.bytesWritten`
-<!-- YAML
-added: v0.4.7
--->
+It is possible to use an {AbortSignal} to cancel an `fsPromises.writeFile()`.
+Cancelation is "best effort", and some amount of data is likely still
+to be written.
 
-The number of bytes written so far. Does not include data that is still queued
-for writing.
+```js esm
+import { writeFile } from 'fs/promises';
 
-### `writeStream.path`
-<!-- YAML
-added: v0.1.93
--->
+try {
+  const controller = new AbortController();
+  const { signal } = controller;
+  const data = new Uint8Array(Buffer.from('Hello Node.js'));
+  fs.writeFile('message.txt', data, { signal });
```

Comment:
```suggestion
  await writeFile('message.txt', data, { signal });
```

Reply:
Likewise, the lack of await here is intentional

Reply:
There's still the typo `fs.writeFile` -> `writeFile` to fix though.

Reply:
Marking it as unresolved as this needs fixing before landing: you need to either import the default export above (`import fs from 'fs/promise'`) or fix this line (see suggestion above).